docs/spec/v1/gitrepositories.md

```diff
@@ -296,6 +297,51 @@ must follow this format:
 ```
 https://dev.azure.com/{your-organization}/{your-project}/_git/{your-repository}
 ```
+#### GitHub
+
+The `github` provider can be used to authenticate to git repositories using
+[GitHub Apps](https://docs.github.com/en/apps/overview).
+
+##### Pre-requisites
+
+- [Register]((https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app))
+  the GitHub App with the necessary permissions and [generate a private
+  key](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/managing-private-keys-for-github-apps)
+  for the app. 
+
+- [Install](https://docs.github.com/en/apps/using-github-apps/installing-your-own-github-app)
+  the app in the organization/account configuring access to the necessary
+  repositories.
+
+##### Configure GitHub App secret
+
+The GitHub App information is specified in `.spec.secretRef` in the format
+specified below:
+
+* Get the App ID from the app settings page at
+  `https://github.com/settings/apps/<app-name>`. 
+* Get the App Installation ID from the app installations page at
+`https://github.com/settings/installations`. Click the installed app, the URL
+will contain the installation ID
+`https://github.com/settings/installations/<installation-id>`. For
+organizations, the first part of the URL may be different, but it follows the
+same pattern.
+* The private key that was generated in the pre-requisites.
+
+```yaml
+apiVersion: v1
+kind: Secret
```

Comment:
We need to implement `flux create secret githubapp` and mention it here as a way to generate this secret.

Reply:
Added support for this in https://github.com/fluxcd/flux2/pull/5103, updated docs to show usage of generating the secret using `flux create secret githubapp` command